calcom/cal.com
# Recurring event improvements

## What does this PR do?

Thanks @hariombalhara for your feedback. Here are the points you mentioned to check one by one.

- [X] We should have a loader on Confirm button as recurring event takes long time to book as there are many requests being sent. **FIXED**
- [X] Rejected Recurring Booking still shows up in Upcoming. Non recurring booking if cancelled or rejected doesn't show up. It shows up in cancelled tab. **FIXED**
- [X] A recurring booking done for same day doesn't show up in upcoming tab. Though it does come up in Recurring Tab. Tested a non recurring booking and it shows up fine in upcoming tab. **FIXED**
 - [X] "Rejecting All" recurring bookings doesn't close the popup. **FIXED**
 - [X] Cancel all remaining events from a confirmed recurring event from recurring tab
 - [X] Showing "X events remaining" from a confirmed recurring event from recurring tab
 - [X] "Every X weeks for Y weeks" was revisited and changed to "Every X weeks for Y occurrences"
<img width="414" alt="Mouse_Highlight_Overlay" src="https://user-images.githubusercontent.com/467258/173128253-fbdd491c-2491-42c1-a4c4-17129e20948d.png">

## Type of change


## How should this be tested?

 - Create recurring events and see them show up in `Upcoming` or `Recurring` depending on confirmation status. Rejecting a whole series from `Upcoming`.

> **Loom**: https://www.loom.com/share/bb3416c289884fc3905a78bca11d8a09
